test-fixture/src/sim/connection.rs

```diff
@@ -82,15 +82,17 @@ impl ConnectionNode {
 
     pub fn default_client(goals: impl IntoIterator<Item = Box<dyn ConnectionGoal>>) -> Self {
         Self::new_client(
-            ConnectionParameters::default().pmtud(true),
+            // Simulator logic does not work with multi-packet MLKEM crypto flights.
+            ConnectionParameters::default().pmtud(true).mlkem(false),
```

Comment:
Do we make use of the simulator in any relevant unit tests? If so, given that `mlkem` is the default, we should as well run those tests with `mlkem`.

Reply:
I'd be happy to see the simulator fixed so that MLKEM works.

Reply:
https://github.com/mozilla/neqo/issues/2410